app/file.go

```diff
@@ -360,7 +360,7 @@ func MigrateFilenamesToFileInfos(post *model.Post) []*model.FileInfo {
 
 func GeneratePublicLink(siteURL string, info *model.FileInfo) string {
 	hash := GeneratePublicLinkHash(info.Id, *utils.Cfg.FileSettings.PublicLinkSalt)
-	return fmt.Sprintf("%s%s/public/files/%v/get?h=%s", siteURL, model.API_URL_SUFFIX_V3, info.Id, hash)
+	return fmt.Sprintf("%s/files/%v/public?h=%s", siteURL, info.Id, hash)
```

Comment:
actually, I did such backwards compatibility in the earlier [PR# 5607](https://github.com/mattermost/platform/pull/5607/files) but was just repeated the function here for test purpose only. I intend to rebase once PR# 5607 is merged.
I've seen the build failure of this PR but not sure if it has something to do with the changes I've made. I'll take a look later. thanks